java/ql/src/Security/CWE/CWE-352/SpringCSRFProtection.qhelp

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE qhelp SYSTEM "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>When a web server is designed to receive a request from a client without any mechanism 
+for verifying that it was intentionally sent, then it might be possible for an attacker 
+to trick a client into making an unintentional request to the web server which will be treated 
+as an authentic request. This can be done via a URL, image load, XMLHttpRequest, etc. and can 
+result in exposure of data or unintended code execution.</p>
+</overview>
+
+<recommendation>
+<p>Cross-Site Request Forgery (CSRF) protection is enabled by default in Spring with Java 
+configuration. It's recommended to not disable this.</p>
+</recommendation>
+
+<example>
+<p>The following example shows the Spring Java configuration with CSRF protection disabled.</p>
```

Comment:
Possibly worth either mentioning how to correct the bad Java configuration, or adding something like this: "This type of configuration should only be used if you are creating a service that is used only by non-browser clients."

Reply:
To correct this, one should not invoke csrf.disable() at all. I'll add: "This type of configuration should only be used if you are creating a service that is used only by non-browser clients."